### ISSUE TITLE
boost 1.81.0 build error on windows

### ISSUE THREAD

**Issue by egorpugin** [boostorg/json#820]
vs2022 17.4.3.

```
When executing: [boost.json-1.81.0]/src/src.cpp
D:\temp\29\.sw\src\2ccd1336e07b\json-31b10c1b70556dfee414ef12f83d85e7d6443508\include\boost/json/impl/value.ipp(403): error C2589: '(': illegal token on right side of '::'
D:\temp\29\.sw\src\2ccd1336e07b\json-31b10c1b70556dfee414ef12f83d85e7d6443508\include\boost/json/impl/value.ipp(404): error C2062: type 'unknown-type' unexpected
D:\temp\29\.sw\src\2ccd1336e07b\json-31b10c1b70556dfee414ef12f83d85e7d6443508\include\boost/json/impl/value.ipp(404): error C2059: syntax error: ')'
```

Probably `WIN32_LEAN_AND_MEAN` or `NOMINMAX` is missing.

**Comment by grisumbras** [comment]
Indeed, the problem is because of the use of std::min.

**Comment by egorpugin** [comment]
I've added NOMINMAX into my build as the less intrusive, it works.

**Comment by grisumbras** [comment]
There will probably be a patch very soon where this is fixed.

**Comment by grisumbras** [comment]
Fixed in #821. A patch for Boost release incoming.
